designer/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/RemoteEnvironment.scala

```diff
@@ -205,15 +222,48 @@ trait StandardRemoteEnvironment extends FailFastCirceSupport with RemoteEnvironm
         processName,
         isFragment
       )
-    invokeForSuccess(
-      HttpMethods.POST,
-      List("migrate"),
-      Query.Empty,
-      HttpEntity(migrateScenarioRequest.asJson.noSpaces),
-      List.empty
-    )
+
+    val f: EitherT[Future, NuDesignerError, Unit] = for {
+      remoteNuVersion <- EitherT(fetchRemoteNuVersion)
+
+      localNuVersion = BuildInfo.version
+
+      versionComparisonResultE = migrationApiAdapterService.compareNuVersions(localNuVersion, remoteNuVersion.value)
+      versionComparisonResult <- EitherT(Future.successful(versionComparisonResultE))
+
+      decidedMigrateScenarioRequest = migrationApiAdapterService.decideMigrationRequestDto(
+        migrateScenarioRequest,
+        versionComparisonResult
+      )
+
+      versionTag = decidedMigrateScenarioRequest match {
+        case _: MigrateScenarioRequestV1_14 => "V1_14"
+        case _: MigrateScenarioRequestV1_15 => "V1_15"
+      }
+
+      res <- EitherT(
+        invokeForSuccess(
+          HttpMethods.POST,
+          List("migrate"),
+          Query.Empty,
+          HttpEntity(decidedMigrateScenarioRequest.asJson.noSpaces),
+          List(
+            RawHeader("X-MigrateDtoVersion", versionTag)
```

Comment:
could you please explain why we need it?

Reply:
Its only relevant for testing purposes, see `StandardRemoteEnvironmentSpec`. Thanks to that I am able to test the version sent was proper one.

Reply:
why mocking version endpoint is a problem?

Reply:
It's not a problem. I just wanted to test more. But it is not so relevant. I can change that one.